### PR TITLE
Adds support for IH when using (i)loc_min(max)

### PR DESCRIPTION
**Pull request by chaburkland**
@flexatone What are your thoughts on this solution to #937 ?

It's marked as a draft now because I didn't want to spend time implementing all the necessary unit tests if you reject the approach outright

If you like it, then I'll get it properly PR ready

### CODE REVIEW

**Comment by chaburkland** [comment]
Refs #937

**Comment by flexatone** [comment]
Thanks, @chaburkland , for this PR: this needed to be addressed! 

I see that you are returning a `Frame` when calling `loc_max` on a  `Frame` with `IndexHierarchy`, but returning a `Series` when calling `loc_max` on a `Frame` with a 1D index. Is this too surprising? We could always return a `Series`; in the case of an `IndexHierarchy`, the Series could be one of tuples for the IH labels, which makes using those values for selection in the same IH straight forward. 

One advantage of what you have done is that the IH name become the column labels: was that your motivation to going to a Frame?

**Comment by chaburkland** [comment]
@flexatone I think you're right that getting back a Frame is a bit surprising. Let me fix it :upside_down_face:

File: static_frame/core/frame.py [hunk 1]
```diff
@@ -7378,6 +7378,35 @@ def sample(self,
...
+
+        if values.ndim == 1:
+            post = values
+        else:
+            post = np.fromiter(
+                map(tuple, values),
+                count=len(values),
+                dtype=object,
+            )
+            post.flags.writeable = False
```
Comment (flexatone) [inline static_frame/core/frame.py:7395]: I think you can use `arraykit.array_to_tuple_array` for an optimized version of this move.

**Review by flexatone — CHANGES_REQUESTED** [review]
Thanks for this PR! Just one opportunity here to use an `arraykit` function.

**Review by flexatone — APPROVED** [review]
Thanks!

**Pull request merged**